qemu/tests/power_htm.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_package
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Run htm cases:
+    Case one
+    1) Download unit test suite and configure it
+    2) Run kvm test on host
+    3) Check host is still available
+
+    Case two
+    1) Download test application in the guest
+    2) Run it in the guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    if params["unit_test"] == "yes":
+        error_context.context("Prepare unit test on host", logging.info)
+        cmds = []
+        cmds.append(params["get_htm"])
+        cmds.append(params["compile_htm"])
+        for cmd in cmds:
+            s, o = process.getstatusoutput(cmd, timeout=3600)
+            if s:
+                test.error("Failed to run cmd '%s', output: %s" % (cmd, o))
+        error_context.context("Run htm unit test on host", logging.info)
+        s, o = process.getstatusoutput(params["run_test"], timeout=3600)
+        if s:
+            test.fail("Run htm unit test failed, output: %s" % o)
+        # Make sure if host is available by do commands on host
+        if process:
+            status, output = process.getstatusoutput('rm -rf %s'
+                                                     % params["htm_dir"])
+        if status:
+            test.fail("Please check host's status: %s" % output)
+        utils_misc.verify_dmesg()
+    else:
+        check_exist_cmd = params["check_env"]
+        s, o = process.getstatusoutput(check_exist_cmd, timeout=3600)
+        if s:
+            test.error("Please check htm is supported or not by '%s', output: %s"
+                       % (check_exist_cmd, o))
+        vm = env.get_vm(params["main_vm"])
+        session = vm.wait_for_login()
+        pkgs = params["depends_pkgs"].split()
+        if not utils_package.package_install(pkgs, session):
+            test.cancel("Install dependency packages failed")
+        session.cmd(params["check_existed"])
+        download_htm_demo = params["download_htm_demo"]
```

Comment:
This is also what I want to say. Without internal `htm_demo`, better replace it with a public link in upstream, it can be any test suite for the guest, or comment it in the cfg file to let users configure it at first.

Reply:
Thanks for your advice so I will add it to the cfg but comment it for user. Thanks.